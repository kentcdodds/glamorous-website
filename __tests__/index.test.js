/* global it, expect, describe */

import React from 'react'
import renderer from 'react-test-renderer'
import Index from '../pages/index'

describe('Testing Index', () => {
  it('Renders!', () => {
    const component = renderer.create(<Index/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
