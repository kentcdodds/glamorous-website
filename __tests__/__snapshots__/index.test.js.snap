// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow 1`] = `
<layout
  pathname=""
>
  <glamorous.Div
    margin={20}
  >
    <glamorous.Div
      margin="0 auto"
      maxWidth={800}
      textAlign="center"
    >
      <GlamorousLogo
        margin="0 auto"
      />
      <glamorous(h1)>
        Maintainable CSS with React
      </glamorous(h1)>
    </glamorous.Div>
  </glamorous.Div>
  <glamorous(div)>
    <CodePreview
      code="const Button = glamorous.a({
      border: '1px solid #F67982',
      width: '11em',
      padding: '0.7em 0',
      textDecoration: 'none',
      borderRadius: 4,
      display: 'inline-block',
      margin: '.5em 1em',
      transition: 'all .3s',
    })
    
    const solidColors = {
      backgroundColor: '#ED5C70',
      color: '#fff',
    }
    
    const transparentColors = {
      backgroundColor: 'rgba(255, 255, 255, 0.5)',
      color: '#ED5C70',
    }
    
    const Primary = glamorous(Button)(
      solidColors,
      {
        ':hover': transparentColors,
      }
    )
    
    const Secondary = glamorous(Button)(
      transparentColors,
      {
        ':hover': solidColors,
      }
    )
    
    render(
      <glamorous.Div textAlign=\\"center\\">
        <Primary href=\\"https://github.com/paypal/glamorous\\">
          GitHub
        </Primary>
        <Secondary href=\\"http://kcd.im/glamorous-help\\">
          Try It
        </Secondary>
      </glamorous.Div>
    )"
    />
    <glamorous(div)>
      npm install --save glamorous react glamor prop-types
    </glamorous(div)>
  </glamorous(div)>
  <glamorous(glamorous(Anchor))
    href="/guides"
  >
    Click Here to get started
  </glamorous(glamorous(Anchor))>
</layout>
`;
